File: api/tests/sanity/openjdk/TestRuntime.java
package sanity;

class SomeClass
{
	public void print()
	{
		System.out.println("Hello from inner class");
	}
}

public class TestRuntime
{
	private TestRuntime(){}

	public static void helloWorld()
	{
		System.out.println("Hello World - From Java");
	}

	public static void returnsAnError() throws Exception
	{
		System.out.println("Going to throw exception");
		throw new Exception("Returning an error");
	}

	public static float divIntegers(int x, int y) throws ArithmeticException
	{
		if(y == 0)
			throw new ArithmeticException("Divisor is 0");

		return (float)x / (float)y;
	}

	public static String joinStrings(String[] arr)
	{
		System.out.println("joining strings");
		return String.join(",", arr);
	}

	public static final int fiveSeconds = 5;
	public static void waitABit(int seconds) throws InterruptedException
	{
		System.out.printf("Sleeping for %d seconds\n", seconds);
	}

	public static SomeClass[] getSomeClasses()
	{
		return new SomeClass[]{new SomeClass(), new SomeClass(), new SomeClass()};
	}

	public static void expectThreeSomeClasses(SomeClass[] arr)
	{
		if(arr.length != 3)
			throw new IllegalArgumentException("Array length is not 3");
	}

	public static void expectThreeBuffers(byte[][] buffers)
	{
		if(buffers.length != 3)
			throw new IllegalArgumentException("Buffers length is not 3");
	}

	public static byte[][] getThreeBuffers()
	{
		byte[][] buffers = new byte[3][];
		for(int i = 0; i < 3; i++)
		{
			buffers[i] = new byte[]{1, 2, 3};
		}
		return buffers;
	}
}

File: api/tests/sanity/go/TestRuntime.go
package TestRuntime

import (
	"strings"
	"time"
	"fmt"
)

func HelloWorld() {
	println("Hello World, From Go!")
}

func ReturnsAnError(){
	panic("An error from ReturnsAnError")
}

func DivIntegers(x int, y int) float32{

	if y == 0{
		panic("Divisor is 0")
	}

	return float32(x) / float32(y)
}

func JoinStrings(arrs []string) string{
	return strings.Join(arrs, ",")
}

const FiveSeconds = time.Second*5
func WaitABit(d time.Duration) error{
	fmt.Printf("Waiting for %v\n", d)
	return nil
}

type TestMap struct{
	m map[string]interface{}
	Name string
}

func NewTestMap() *TestMap{
	return &TestMap{ 
		m: make(map[string]interface{}),
		Name: "TestMap Name",
	}
}

func (this *TestMap) Set(k string, v interface{}){
	fmt.Printf("Setting: %T\n", v)
	this.m[k] = v
}

func (this *TestMap) Get(k string) interface{}{
	v := this.m[k]
	fmt.Printf("Getting: %T\n", v)
	return v
}

func (this *TestMap) Contains(k string) bool{
	_, found := this.m[k]
	return found
}
